fv3core/stencils/saturation_adjustment.py

```diff
@@ -246,47 +209,42 @@ def melt_cloud_ice(
 
 @gtscript.function
 def minmax_tmp_h20(qa, qb):
-    tmpmax = qb if qb > 0.0 else 0.0
-    tmp = -qa if -qa < tmpmax else tmpmax
-    return tmp
+    tmpmax = max(qb, 0.0)
+    return min(-qa, tmpmax)
 
 
-# Fix negative snow with graupel or graupel with available snow.
-# TODO Fix so only compute tmp when it is needed
 @gtscript.function
 def fix_negative_snow(qs, qg):
-    tmp = minmax_tmp_h20(qg, qs)
     if qs < 0.0:
         qg = qg + qs
         qs = 0.0
     elif qg < 0.0:
+        tmp = minmax_tmp_h20(qg, qs)
```

Comment:
it really is nice we don't have to define this for every branch when it's only used in one.